core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -731,8 +731,12 @@ private[kafka] abstract class Acceptor(val socketServer: SocketServer,
         s" sendBufferSize [actual|requested]: [${socketChannel.socket.getSendBufferSize}|$sendBufferSize]" +
         s" recvBufferSize [actual|requested]: [${socketChannel.socket.getReceiveBufferSize}|$recvBufferSize]")
       true
-    } else
+    } else{
+      // connection was rejected (likely due to processor shutdown) - close the socket.
+      val listenerName = ListenerName.normalised(endPoint.listener)
+      connectionQuotas.closeChannel(this, listenerName, socketChannel)
```

Comment:
```suggestion
      connectionQuotas.closeChannel(this, normalised(endPoint.listener), socketChannel)
```
might consider cohesion.

Reply:
thank you for the suggestion, have made the change